wallet/src/de/schildbach/wallet/ui/LockScreenActivity.kt

```diff
@@ -545,4 +552,24 @@ open class LockScreenActivity : SecureActivity() {
             inputManager?.hideSoftInputFromWindow(token, 0)
         }
     }
+
+    fun registerLockScreenDeactivated(listener: () -> Unit) {
+        lockScreenDeactivatedListeners.add(listener)
+    }
+
+    fun unregisterLockScreenDeactivated(listener: () -> Unit) {
+        lockScreenDeactivatedListeners.remove(listener)
+    }
 }
+
+fun Fragment.registerLockScreenDeactivated(listener: () -> Unit) {
+    if (activity is LockScreenActivity) {
+        (activity as LockScreenActivity).registerLockScreenDeactivated(listener)
+    }
+}
+
+fun Fragment.unregisterLockScreenDeactivated(listener: () -> Unit) {
+    if (activity is LockScreenActivity) {
+        (activity as LockScreenActivity).unregisterLockScreenDeactivated(listener)
+    }
+}
```

Comment:
This should be possible to do with `LockScreenAware` interface, which will provide methods to override

Reply:
I will refactor this in the next PR.  `LockScreenAware` may need modification.